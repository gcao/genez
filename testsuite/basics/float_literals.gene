#!/usr/bin/env gene run

# Test: Float literal parsing and operations

# Basic float literals
(print 3.14)
(print 0.5)
(print 42.0)

# Float arithmetic
(print (+ 1.5 2.5))
(print (- 10.0 3.7))
(print (* 2.5 4.0))
(print (/ 10.0 3.0))

# Float comparisons
(print (< 3.14 4.0))
(print (> 5.5 3.3))
(print (== 2.0 2.0))
(print (== 2.0 2.1))

# Mixed int/float operations
(print (+ 1 2.5))
(print (* 3 1.5))