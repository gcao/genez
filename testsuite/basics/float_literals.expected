3.14
0.5
42
4
6.3
10
3.3333333333333335
true
true
true
false
3.5
4.5